### PR TITLE
Added Mission/Commissions/Event

### PR DESCRIPTION
**Pull request by RedDeadDepresso**
I tested for some time and everything works fine. I'll briefly explain how it works:

1. Read config from MCE/config.json and load the queue
2. Update the dashboard
3. For each stage calculate the "realistic count" ie the number of sweeps possible based on the current ap
4. Sweep the stage for the realistic count
5. Save the difference between realistic count and the number of sweep wanted by the user. If the difference is 0 remove the stage from the queue.
6. Keep doing this until realistic count is 0. Recharge AP if enabled and repeat the process again. Otherwise terminate Mission.

Also, I tried to use the functions write_file and read_file from module/config/config.py but they don't work. I think I add enough error handling so it shouldn't be a problem.

Changes made:
1. Created  mission_list.py, a version of task/stage/stage_list.py but with a shorter swipe range, use string comparison instead of integer and regex that also matches stages where the last character is a capital letter for example 3-A. All these changes were made because of Mission.

Optional Optimisations :
This is not necessary but would be nice,
- add more error handling in select_area() in mission/ui.py. I used a try except block but it gonna be stuck for a while if it doesn't find the area.

TODO:
- Add event quest switch and event_page for ui_ensure. Unfortunately, there are no active events in Global.
- Make AAS generate a MCE/config.json if it doesn't exist. The one already in this pull request is the default config but I'm not sure where I should put the code to generate it in AAS if not found.

### CODE REVIEW

**Comment by RedDeadDepresso** [comment]
This is the latest fork of dev with Mission/Commissions/Event and with the updated assets of tactical challenge from the master branch. You don't need to worry about file conflicts.

File: tasks/task/task.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def run(self):
                     self.device.click(CLAIM)
                     logger.info("Click Claim")
                     continue
-                if self.match_color(CLAIMED) and self.match_color(CLAIMED_ALL):
+                if not self.match_color(CLAIMED) and not self.match_color(CLAIMED_ALL):
```
Comment (TheFunny) [inline tasks/task/task.py:26]: The original one should be right? `CLAIMED` is the gray button.

Comment (RedDeadDepresso) [reply tasks/task/task.py]: Sorry, I got confused with the CLAIM and CLAIM_ALL buttons.

**Comment by TheFunny** [comment]
For generating MCE config, I don't know where to put the code in AAS either. Maybe just let MCE generate the config itself because user have to open that first if they want to use the mission function? I consider this as a temporary solution and the ultimate solution is to implement mission option in AAS gui (I'll try when I'm free).

**Comment by RedDeadDepresso** [comment]
Done

**Comment by RedDeadDepresso** [comment]
What is the maximum number of sweeps possible before triggering too many click exceptions? I can set the limit in MCE Manager.

**Comment by RedDeadDepresso** [comment]
What do you think about this solution to close random popup in main? From my tests it seems to work.

**Comment by TheFunny** [comment]
> What is the maximum number of sweeps possible before triggering too many click exceptions? I can set the limit in MCE Manager.

The limit of clicks can be bypassed if you use the function to clean the count of clicks (I've used that in the cafe module when tapping students), but make sure to set another limit such as timeout.

> What do you think about this solution to close random popup in main? From my tests it seems to work. This could also work for random popups when entering the game, maybe you could keep pressing the emulator back button until main after Menu button disappears? It will just show option is not available and it won't trigger too many click exceptions since you are not clicking on the screen.
> 
> Also, I discovered why Tasks wasn't working for EN. It was because CLAIMED button was different.

That is actually a feasible solution to solve the popups in Global server. I can complete that in the login process later.
For claimed button, I just found that the already claimed one and the not claimed one are different from each other, but I haven't got time to fix that. Anyway, thanks for solving this.

**Comment by RedDeadDepresso** [comment]
Hi, I think you can merge.

For sweeps I would prefer if you could do it yourself since you wrote it and I'm more likely to make mistakes.

The shop seems to work well in Global, just make sure to test it in JP.

At this point, all you have to do is fix momotalk, improve login and sweep for the mission and add ui_ensure for event. If you have any problems with MCE Manager I can try to fix it.

**Comment by RedDeadDepresso** [comment]
Ok, this is the final fix for task. The claim button has two states: CLAIMED when you have not completed the 9 daily tasks and COMPLETE after you complete them.

**Review by TheFunny — APPROVED** [review]

**Comment by TheFunny** [comment]
Thanks. I'll check the functions for JP when I have time.

**Pull request merged**